src/main.ts

```diff
@@ -62,11 +65,25 @@ const getTargetVersion = async (
 ): Promise<string | null> => {
 	return await balena.models.os
 		.getSupportedOsUpdateVersions(deviceType, deviceVersion)
-		.then((versions) => {
-			if (versions.recommended) {
-				return versions.recommended;
-			} else {
+		.then((osUpdateVersions) => {
+			if (userTargetVersion === '') {
+				console.log(
+					'HUP_TARGET_VERSION must be set to perform automatic updates.',
+				);
 				return null;
+			} else {
+				if (
+					userTargetVersion === 'recommended' ||
```

Comment:
```suggestion
					['recommended','latest'].includes(userTargetVersion)
```
I sometimes find this syntax easier to read, but it's up to you.